src/reportengine/environment.py

```diff
@@ -117,5 +120,4 @@ def ns_dump_description(cls):
         )
 
     def ns_dump(self):
-        return {k: getattr(self, k) for k in self.ns_dump_description()}
-
+        return {k: getattr(self, k) for k in self.ns_dump_description() if getattr(self, k)}
```

Comment:
The condition should be at least `... if getattr(self, k) is not None` because one might want to dump attributes that are not logically true.  A slightly more refined way of doing things, in case you decide we also might want `None` items is to use a singleton as a default as in:

```
class EMPTY:
    pass
...

def __init__(self, *, output=EMPTY, formats=('pdf',)

... if getattr(self, k) is not EMPTY
```

Using a class is worse than `EMPTY = object()`, which would achieve a similar effect because it is more expensive, but better because it prints nicer. If at some point you want it to play well with type checkers, you would likely do:

```
class _EMPTY_TYPE:
    pass
EMPTY = _EMPTY_TYPE()